vaccine-standby/functions/setup.js

```diff
@@ -0,0 +1,507 @@
+        transitions: [
+          {
+            next: 'Goodbye',
+            event: 'incomingMessage',
+          },
+          {
+            event: 'timeout',
+          },
+          {
+            event: 'deliveryFailure',
+          },
+        ],
+        properties: {
+          offset: {
+            x: 60,
+            y: 3210,
+          },
+          service: '{{trigger.message.InstanceSid}}',
+          channel: '{{trigger.message.ChannelSid}}',
+          from: '{{flow.channel.address}}',
+          body: 'What is your preferred language?\n1. English\n2. Español\n3. 繁體中文\n\nReply"1", "2", or "3"',
+          timeout: '3600',
+        },
+      },
+      {
+        name: 'ZipCode',
+        type: 'send-and-wait-for-reply',
+        transitions: [
+          {
+            next: 'EssentialWorker',
+            event: 'incomingMessage',
+          },
+          {
+            event: 'timeout',
+          },
+          {
+            event: 'deliveryFailure',
+          },
+        ],
+        properties: {
+          offset: {
+            x: 360,
+            y: 1120,
+          },
+          service: '{{trigger.message.InstanceSid}}',
+          channel: '{{trigger.message.ChannelSid}}',
+          from: '{{flow.channel.address}}',
+          body: 'What is your 5-digit zip code?',
+          timeout: '3600',
+        },
+      },
+    ],
+    initial_state: 'Trigger',
+    flags: {
+      allow_concurrent_calls: true,
+    },
+  };
+
+  const client = context.getTwilioClient();
+
+  // Deploy Twilio Studio Flow
+  function deployStudio(formattedFlow) {
+    return new Promise((resolve, reject) => {
+      client.studio.flows
+        .create({
+          commitMessage: 'Code Exchange automatic deploy',
+          friendlyName: 'Vaccine Standby Intake',
+          status: 'published',
+          definition: formattedFlow,
+        })
+        .then((flow) => {
+          resolve(flow.webhookUrl);
+        })
+        .catch((err) => reject(err.details));
+    });
+  }
+
+  function getPhoneNumberSid() {
+    return new Promise((resolve, reject) => {
+      client.incomingPhoneNumbers
+        .list({ phoneNumber: context.TWILIO_PHONE_NUMBER, limit: 20 })
+        .then((incomingPhoneNumbers) => {
+          const n = incomingPhoneNumbers[0];
+          resolve(n.sid);
+        })
+        .catch((err) => reject(err));
+    });
+  }
+
+  function updatePhoneNumberWebhook(studioWebhook, numberSid) {
+    return new Promise((resolve, reject) => {
+      client.incomingPhoneNumbers(numberSid)
+        .update({
+          smsUrl: studioWebhook,
+        })
+        .then(() => {
+          console.log(`Successfully updated ${numberSid} to use ${studioWebhook}.`);
+          resolve('success');
+        })
+        .catch((err) => reject(err));
+    });
+  }
+
+  const webhookUrl = await deployStudio(flowDefinition);
+  const phoneNumberSid = await getPhoneNumberSid();
+  await updatePhoneNumberWebhook(webhookUrl, phoneNumberSid);
```

Comment:
you have catches set up for each of these function calls, but never return those errors via the callback, or check if the previous step returned an error before calling the next. So the `callback(null, 'success')` will trigger every time regardless of failures